scripts/networkSimulator.js

```diff
@@ -0,0 +1,269 @@
+        if (tradeData.isTestingMaxDestAmount) {
+            let info = [tradeData.srcQty, networkFeeBps, tradeData.platformFeeBps];
+            [expectedResult, actualSrcQty] = await nwHelper.calcParamsFromMaxDestAmt(tradeData.srcToken, tradeData.destToken, expectedResult, info, tradeData.maxDestAmount);
+        }
+        console.log(`Loop ${loop}: Execute trade ( ${tradeData.message}, networkFeeBps=${networkFeeBps.toString(10)} ),
+            srcQty=${tradeData.srcQty.toString(10)}, srcDecimals=${tradeData.srcDecimals},
+            dstQty=${expectedResult.actualDestAmount.toString(10)}, dstDecimals=${tradeData.destDecimals},
+            platformFee=${expectedResult.platformFeeWei.toString(10)}, networkFee=${expectedResult.networkFeeWei.toString(10)}, accountedFeeBps=${expectedResult.feePayingReservesBps.toString(10)},
+            rateWithNetworkFee=${expectedResult.rateWithNetworkFee.toString(10)},
+            rateWithAllFees=${expectedResult.rateWithAllFees.toString(10)}`);
+        await network.tradeWithHintAndFee(
+            proxyAddress,
+            srcAddress,
+            tradeData.srcQty,
+            destAddress,
+            taker,
+            tradeData.maxDestAmount,
+            tradeData.minConversionRate,
+            taker, // platform wallet
+            tradeData.platformFeeBps,
+            tradeData.hint,
+            {from: networkProxy, value: callValue, gasPrice: new BN(0)}
+        );
+        await nwHelper.compareBalancesAfterTrade(tradeData.srcToken, tradeData.destToken, actualSrcQty,
+            initialReserveBalances, initialTakerBalances, expectedResult, taker, proxyAddress);
+        return true;
+    }
+    
+    console.log(`Loop ${loop}: Getting zero rates for trade`);
+    return false;
+}
+
+async function updateRatesForReserves(reserveInstances, tokens, accounts) {
+    let ethInit = new BN(100).mul(new BN(10).pow(new BN(18)));
+    let ethSender = 0;
+    for(const [key, value] of Object.entries(reserveInstances)) {
+        let reserve = value.instance;
+        // deposit more eth if needed
+        await reserve.withdrawAllEth({from: accounts[ethSender]});
+        await Helper.sendEtherWithPromise(accounts[ethSender], reserve.address, ethInit);
+        let val = TradeParamGenerator.getRandomInt(1, (ethSender + 1) * 10);
+        let tokensPerEther = precisionUnits.mul(new BN(val));
+        let ethersPerToken = precisionUnits.div(new BN(val));
+        for(let i = 0; i < tokens.length; i++) {
+            let token = tokens[i];
+            await reserve.withdrawAllToken(token.address);
+            let initialTokenAmount = new BN(2000000).mul(new BN(10).pow(new BN(await token.decimals())));
+            await token.transfer(reserve.address, initialTokenAmount);
+            await reserve.setRate(token.address, tokensPerEther, ethersPerToken);
+        }
+        ethSender++;
+    }
+}
+
+async function zeroNetworkBalance(network, tokens, admin) {
```

Comment:
better move this function to helper
so we don't duplicate code.

Reply:
Ok